data/etl/teleicare_history/extractor.py

```diff
@@ -451,7 +489,29 @@ def create_declarations_from_teleicare_history(company_ids=[]):
             latest_ica_version_declaration = declaration_versions.order_by("vrsdecl_numero").last()
             # la déclaration a une version finalisée
             if latest_ica_version_declaration:
-                declaration_creation_date = (
+                try:  # pas possible d'utiliser update_or_create
+                    declaration = Declaration.objects.get(
+                        siccrf_id=ica_complement_alimentaire.cplalim_ident,
+                        company=EtablissementToCompanyRelation.objects.get(
+                            siccrf_id=ica_complement_alimentaire.etab_id
+                        ).company,
+                    )
+                    if not rewrite_existing:
+                        continue
+                except Declaration.DoesNotExist:
+                    declaration = Declaration(
+                        siccrf_id=ica_complement_alimentaire.cplalim_ident,
+                        company=EtablissementToCompanyRelation.objects.get(
+                            siccrf_id=ica_complement_alimentaire.etab_id
+                        ).company,
+                    )
+                declaration_attrs = compute_declaration_attributes(
+                    ica_complement_alimentaire, latest_ica_declaration, latest_ica_version_declaration
+                )
+                for key, value in declaration_attrs.items():
```

Comment:
Si je comprends bien ici on itère au lieu de fare `declaration.update(**declaration_attrs)` car on ne veut pas encore déclencher une sauvegarde c'est ça ?

Reply:
Tout à fait. Il y a un appel à `.save()` après qui doit être effectué dans le contexte `suppress_autotime`
C'est aussi la raison pour laquelle je n'ai pas utilisé `update_or_create` de Django